src/main/scala/xiangshan/frontend/ifu/Bundles.scala

```diff
@@ -70,21 +72,62 @@ class FetchBlockInfo(implicit p: Parameters) extends IfuBundle {
   val target:         PrunedAddr  = PrunedAddr(VAddrBits)
   val instrRange:     UInt        = UInt(FetchBlockInstNum.W)
   val rawInstrEndVec: UInt        = UInt(FetchBlockInstNum.W)
-  val pcHigh:         UInt        = UInt((VAddrBits - PcCutPoint).W)
   val pcHighPlus1:    UInt        = UInt((VAddrBits - PcCutPoint).W)
   val fetchSize:      UInt        = UInt(log2Ceil(FetchBlockInstNum + 1).W)
   val identifiedCfi:  Vec[Bool]   = Vec(FetchBlockInstNum, Bool())
+  def pcHigh:         UInt        = startVAddr(VAddrBits - 1, PcCutPoint)
+  def getBasicBlockIdx(pc: PrunedAddr, start: PrunedAddr): UInt = {
+    val byteOffset = (pc - start).toUInt
+    (byteOffset - instBytes.U)(FetchBlockInstOffsetWidth, instOffsetBits)
+  }
+  def fromFtqRequest(ftqFetch: FetchRequestBundle): FetchBlockInfo = {
```

Comment:
```suggestion

  def pcHigh:         UInt        = startVAddr(VAddrBits - 1, PcCutPoint)

  def getBasicBlockIdx(pc: PrunedAddr, start: PrunedAddr): UInt = {
    val byteOffset = (pc - start).toUInt
    (byteOffset - instBytes.U)(FetchBlockInstOffsetWidth, instOffsetBits)
  }

  def fromFtqRequest(ftqFetch: FetchRequestBundle): FetchBlockInfo = {
```
Leave empty line between methods.